### ISSUE TITLE
NXOS- 'show feature' regex error

### ISSUE THREAD

**Issue by bjwebster** [CiscoTestAutomation/genieparser#792]
The NXOS 'show feature' parser does not parse a routing protocol feature if it is enabled but not running.  For example, if EIGRP is not enabled the show output is:
`eigrp                 1         disabled`

when is is enabled but not running, the output is:
`eigrp                 1         enabled (not-running)`

and when it is both enabled and running, the output is:
`eigrp                 1         enabled`

The parser produces an output for EIGRP for the disabled and enabled + running but not for enabled + not running. The feature is missing from the parser output in this case.

I have debugged the parser and I think that the error is in the regex expression.

The existing expression is:
```
(r'^(?P<name>[\w\-]+) +(?P<inst>\d+) '
                             '+(?P<state>\w+)\(?(?P<run>not-running)?\)?$')
```

The modified and working expression is:

`(r'^(?P<name>[\w\-]+) +(?P<inst>\d+) +(?P<state>\w+) *(\(?(?P<run>not-running)?\)?$)')`

**Comment by sowmyadn010501** [comment]
I will be working on your ticket. Kindly give me some time to analyze the issue

**Comment by sowmyadn010501** [comment]
Hi,
can you please provide the complete device output for that regex.
kindly let me know, If you have already found the solution so that you can raise the PR and contribute to the repository.

**Comment by bjwebster** [comment]
[genie_issue#793.txt](https://github.com/CiscoTestAutomation/genieparser/files/13364713/genie_issue.793.txt)

I have attached a file showing the output from the device CLI with the EIGRP feature enabled but not running. Following that is the the output of the parser with the original REGEX expression. You can see that the EIGRP section is missing. The final section shows the output of the parser with the modified REGEX. This now has the EIGRP section. 

I only have access to a device running version 7.3(8)N1(1) so I am not able to tell if this issue affects other versions.

I can raise a PR if you are happy with the limitation above.

**Comment by sowmyadn010501** [comment]
Thank you for the update. Kindly give me some time go through.

**Comment by sowmyadn010501** [comment]
If your script satisfies the existed [expected output](https://github.com/CiscoTestAutomation/genieparser/tree/e79be80e384edef6e629e47d0201faaee9daa67e/src/genie/libs/parser/nxos/tests/ShowFeature/cli/equal) and not throwing any error, then you are good to raise the PR for your changes in the script.

**Comment by bjwebster** [comment]
I think I have found the source of the issue. The _golden_output_output.txt_ in the tests has a slightly different format than the version that I get from the device. 
The _golden_output_output.txt_ version is:

> eigrp                  1          enabled 
> eigrp                  2          enabled(not-running)
> eigrp                  3          enabled(not-running)
> eigrp                  4          enabled(not-running)

Whilst the version that I am getting is:

> eigrp                 1         enabled
> eigrp                 2         enabled (not-running)
> eigrp                 3         enabled (not-running)
> eigrp                 4         enabled (not-running)

It is the space between _enabled_ and _(not-running)_ that is the problem.

I checked the output from a Nexus 9000v in the DevNet Sandbox and it has no space, as per the golden output. It looks like Cisco have changed the format in later versions of software than mine. This means that the parser is correct and does not need to be modified. I am therefore closing this issue.
